src/parsers/utils.hpp

```diff
@@ -52,6 +80,10 @@ namespace se3
     // else return an empty string that will provoke an error in loadPolyhedronFromResource()
     namespace bf = boost::filesystem;
 
+    // In case the mesh_path is such as "file://path/to/mesh", check for /path/to/mesh and return it if it exists
+    if ( bf::exists( bf::path(urdf_mesh_path.substr(6, urdf_mesh_path.size())) ))
+      return std::string(urdf_mesh_path.substr(6, urdf_mesh_path.size()));
```

Comment:
This seems very weak to me. Why don't you check if the path starts with `file://` or `package://` ?

This code may generate unexpected errors like:
- for file `visual/somepath`, we will first check if `/somepath` exists,
- for `collision/somepath`, we will first check if `packageDir/somepath` exists.

By the way, in the standard UNIX file naming, `file://path/to/mesh` is a relative path (`path/to/mesh`) and  `file:///path/to/mesh` is an absolute path (`/path/to/mesh`) so:
- the comment is wrong,
- the substring should start at 7, not 6.